concordium-sdk/src/main/java/com/concordium/sdk/responses/transactionstatus/ContractUpdated.java

```diff
@@ -1,43 +1,75 @@
 package com.concordium.sdk.responses.transactionstatus;
 
-import com.concordium.sdk.transactions.AccountAddress;
-import com.concordium.sdk.transactions.GTUAmount;
+import com.concordium.sdk.transactions.CCDAmount;
 import com.fasterxml.jackson.annotation.JsonCreator;
 import com.fasterxml.jackson.annotation.JsonProperty;
-import com.fasterxml.jackson.annotation.JsonSubTypes;
-import com.fasterxml.jackson.annotation.JsonTypeInfo;
 import lombok.Getter;
 import lombok.ToString;
 
 import java.util.List;
 import java.util.Map;
 import java.util.Objects;
 
+/**
+ * A contract was updated
+ */
 @ToString
 @Getter
 public class ContractUpdated extends TransactionResultEvent {
-    private GTUAmount amount;
+
+    /**
+     * The amount provided
+     */
+    private CCDAmount amount;
+
+    /**
+     * The instigator i.e. the source invoking the contract.
+     */
     private final AbstractAccount instigator;
```

Comment:
Why is this an abstract account, and not an account address?

Reply:
Not sure I understand. The abstract account can either be an account or a contract. The abstract account is a superclass for both of these types.

Reply:
Agree. Will rename it.